maggy/core/executors/Executor.py

```diff
@@ -0,0 +1,96 @@
+#
+#   Copyright 2021 Logical Clocks AB
+#
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+#
+
+from hops.experiment_impl.util import experiment_utils
+
+from maggy.core.executors import trial_executor, dist_executor
+from maggy.core.experiment_driver.OptimizationDriver import OptimizationDriver
+from maggy.core.experiment_driver.AblationDriver import AblationDriver
+from maggy.core.experiment_driver.DistributedDriver import DistributedDriver
+
+
+class Executor:
+    """Wrapper class around the different executor functions.
+
+    Executor functions monkey patch the training function to fit the experiment framework.
+
+    Attributes:
+        exp_driver (Union[OptimizationDriver, AblationDriver, DistributedDriver]): Experiment driver
+            for the patching function.
+    """
+
+    def __init__(self, exp_driver):
+        """Asserts if the driver is of correct and saves the driver.
+
+        Args:
+            exp_driver (Union[OptimizationDriver, AblationDriver, DistributedDriver]): Experiment
+            driver for the patching function.
+        """
+        assert type(exp_driver) in [
+            OptimizationDriver,
+            AblationDriver,
+            DistributedDriver,
+        ], f"Experiment driver type {type(exp_driver)} unsupported by Executor."
```

Comment:
I tend to prefer `isinstance` instead of checking type equality, it's a bit safer since it checks also for inheritence. In this case it doesn't matter, just for the future 
```suggestion
        assert isinstance(exp_driver, [
            OptimizationDriver,
            AblationDriver,
            DistributedDriver,
        ]), f"Experiment driver type {type(exp_driver)} unsupported by Executor."
```